test/es6/proxybugs.js

```diff
@@ -383,6 +383,52 @@ var tests = [
             assert.isTrue(trapCalled);
         }
     },
+    {
+        name: "Extending a proxy with an es6 class",
+        body() {
+            function parent() { this.noTrap = true; }
+            var proxyNoTrap = new Proxy(parent, {});
+            var handler = {
+                construct : function () {
+                    this.other = true;
+                    return { trap: true };
+                }
+            }
+            var proxyWithTrap = new Proxy(parent, handler );
+
+            class NoTrap extends proxyNoTrap
+            {
+                constructor()
+                {
+                    super();
+                    this.own = true;
+                }
+                a () { return true; }
+            }
+
+            class WithTrap extends proxyWithTrap
+            {
+                constructor()
+                {
+                    super();
+                    this.own = true;
+                }
+                a () { return true; }
+            }
+            
+            var notrap = new NoTrap();
+            assert.isTrue(notrap.own);
+            assert.isTrue(notrap.a());
+            assert.isTrue(notrap.noTrap);
+            
+            var withtrap = new WithTrap();
+            assert.isTrue(withtrap.own);
+            assert.throws(withtrap.a);

```

Comment:
Why does this throw? Is it because `withtrap` doesn't connect up to the WithTrap prototype so it doesn't have the `a` method? In that case, should this be `assert.isUndefined` instead of `assert.throws` ?

Reply:
Good point, this seems to be testing incorrect behavior. If this currently throws, then we should leave a commented `assert.isUndefined` and log another bug to follow up.

---
In reply to: [178618653](https://github.com/Microsoft/ChakraCore/pull/4908#discussion_r178618653) [](ancestors = 178618653)

Reply:
That’s correct - when you have a trap it does not connect to the prototype (this felt weird to me but is the same behaviour as jsc, sm and v8)

Could indeed change to assert.isUndefined